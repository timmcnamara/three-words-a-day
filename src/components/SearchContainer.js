import React from "react";
import Button from "./Button";
import TextInput from "./TextInput";

class SearchContainer extends React.Component {
  handleClick(e) {
    e.preventDefault();
    console.log("Submitting....");
  }

  onChange(e) {
    e.preventDefault();
    console.log("change detected");
  }

  render() {
    return (
      <React.Fragment>
        <TextInput placeholder="Enter any word" onChange={this.onChange} />
        <Button onClick={this.handleClick} text="Get it" />
      </React.Fragment>
    );
  }
}

export default SearchContainer;
