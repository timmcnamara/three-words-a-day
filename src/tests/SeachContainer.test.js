import { render, cleanup } from "react-testing-library";
import React from "react";
import SearchContainer from "../components/SearchContainer";
import "jest-dom/extend-expect";

afterEach(cleanup);

// run a snapshot
it("render SearchContainer component", () => {
  const { asFragment } = render(<SearchContainer />);
  expect(asFragment()).toMatchSnapshot();
});

it("onchange function to return ... ", () => {});

it("onClick function ... ", () => {});
