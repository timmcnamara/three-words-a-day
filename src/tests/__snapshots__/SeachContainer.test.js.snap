// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchContainer component 1`] = `
<DocumentFragment>
  <input
    aria-label="input"
    class="sc-bwzfXH eCMyJy"
    placeholder="Enter any word"
  />
  <button
    class="sc-bdVaJa ljraLK"
    data-testid="button-global"
  >
    Get it
  </button>
</DocumentFragment>
`;
